treeherder/webapp/api/runnable_jobs.py

```diff
@@ -80,7 +80,11 @@ def list(self, request, project):
         for label, node in tc_graph.iteritems():
             task_metadata = node['task']['metadata']
             treeherder_options = node['task']['extra']['treeherder']
-            build_platform = treeherder_options['build']['platform']
+            build = treeherder_options.get('build', '')
+            if build != '':
+                build_platform = build.get('platform')
+            else:
+                build_platform = ''
```

Comment:
This can be written as:
```python
build_platform = treeherder_options.get('build', {}).get('platform')
```

Please also add a comment explaining:
(a) why the `build` key only sometimes exists
(b) whether this is permanent or can be removed when changes move through the trains

The commit message also needs the bug number adding (and ideally rewording to be a little clearer, eg "Make RunnableJobs API handle the new taskgraph schema" or similar).

Thanks!